do-task-now.lua

```diff
@@ -0,0 +1,226 @@
+-- do-task-now makes a job involving current selection high priority.
+
+-- place this file in your /df/hack/scripts folder.
+
+local function print_help()
+    print [====[
+
+do-task-now
+===========
+Select anything in the game and execute ``do-task-now`` to make
+a related job high priority. For a building only construct and
+deconstruct jobs are considered. For a work order, all its jobs
+are affected.
+]====]
+end
+
+local utils = require 'utils'
+
+local function getUnitName(unit)
+    local language_name = dfhack.units.getVisibleName(unit)
+    if language_name.has_name then
+        return dfhack.df2console(dfhack.TranslateName( language_name ))
+    end
+    
+    -- animals
+    return dfhack.units.getProfessionName(unit)
+end
+
+local function doJobNow(job)
+    local job_str = dfhack.job.getName(job)
+    if not job.flags.do_now then
+        job.flags.do_now = true
+        print("Made the job " .. job_str .. " top priority")
+    else
+        print("The job " .. job_str .. " is already top priority")
+    end
+    local building = dfhack.job.getHolder(job)
+    if building then
+        print("... at " .. utils.getBuildingName(building))
+    end
+    local unit = dfhack.job.getWorker(job)
+    if unit then
+        print("... by " .. getUnitName(unit) )
+    end
+end
+
+local function doItemJobNow(item)
+    if not item.flags.in_job then
+        qerror(dfhack.items.getDescription(item, 0) .. " must be in a job! (look for 'TSK')")
+    end
+
+    for _, sref in ipairs(item.specific_refs) do
+        if sref.type == df.specific_ref_type.JOB then
+            doJobNow(sref.data.job)
+            return
+        end
+    end
+    print("Couldn't find any job for " .. dfhack.items.getDescription(item, 0))
+end
+
+local function doBuildingJobNow(building)
+    --print('This will attempt to make a job of a building a top priority')
+
+    if #building.jobs > 0
+        and    (
+            building.jobs[0].job_type == df.job_type.ConstructBuilding
+         or building.jobs[0].job_type == df.job_type.DestroyBuilding
+        )
+    then
+        doJobNow(building.jobs[0])
+        return
+    end
+    print("Couldn't find any job for " .. utils.getBuildingName(building))
```

Comment:
maybe clarify here that we couldn't find any ConstructBuilding or DestroyBuilding job